src/float_object.cpp

```diff
@@ -30,6 +30,7 @@ bool FloatObject::eq(Env *env, Value other) {
         return m_double == other.get_fast_integer();
     if (other.is_fast_float())
         return m_double == other.get_fast_float();
+    other.unguard();
```

Comment:
We lose a bunch of benefit of this being an "optimized" method if we call unguard here.